backend-api/src/functions/proxy/proxyHandler.ts

```diff
@@ -0,0 +1,142 @@
+    });
+    return {
+      headers: {
+        "Content-Type": "application/json",
+      },
+      statusCode: 500,
+      body: JSON.stringify({
+        error: "server_error",
+        error_description: "Server Error",
+      }),
+    };
+  }
+
+  const { path } = event;
+  const allowedPaths = ["/async/token", "/async/credential"];
+
+  if (!allowedPaths.includes(path)) {
+    logger.log("UNEXPECTED_PATH", {
+      errorMessage: "Path is not one of the permitted values",
+    });
+    return {
+      headers: {
+        "Content-Type": "application/json",
+      },
+      statusCode: 500,
+      body: JSON.stringify({
+        error: "server_error",
+        error_description: "Server Error",
+      }),
+    };
+  }
+
+  const method = event.httpMethod;
+
+  if (method !== "POST") {
+    logger.log("UNEXPECTED_HTTP_METHOD", {
+      errorMessage: "Method is not an accepted value",
+    });
+    return {
+      headers: {
+        "Content-Type": "application/json",
+      },
+      statusCode: 500,
+      body: JSON.stringify({
+        error: "server_error",
+        error_description: "Server Error",
+      }),
+    };
+  }
+
+  const incomingHeaders = event.headers;
+  const standardisedHeaders = standardiseAndStripApiGwHeaders(incomingHeaders);
+
+  const customAuthHeaderValue =
+    standardisedHeaders["X-Custom-Auth"] ??
+    standardisedHeaders["x-custom-auth"];
+
+  if (customAuthHeaderValue) {
+    standardisedHeaders["Authorization"] = customAuthHeaderValue;
+  }
+
+  const proxyRequestService = dependencies.proxyRequestService();
+  const proxyRequestResult = await proxyRequestService.makeProxyRequest({
+    backendApiUrl: configResult.value.ASYNC_BACKEND_API_URL,
+    body: event.body,
+    path,
+    headers: standardisedHeaders,
+    method: method,
+  });
+  if (proxyRequestResult.isError) {
+    logger.log("PROXY_REQUEST_ERROR", {
+      errorMessage: proxyRequestResult.value.errorMessage,
+    });
+
+    return {
+      headers: {
+        "Content-Type": "application/json",
+      },
+      statusCode: 500,
+      body: JSON.stringify({
+        error: "server_error",
+        error_description: "Server Error",
+      }),
+    };
+  }
+
+  logger.log("COMPLETED");
+  return proxyRequestResult.value;
+}
+
+export const lambdaHandler = lambdaHandlerConstructor.bind(null, dependencies);
+
+const standardiseAndStripApiGwHeaders = (
+  apiGwHeaders: APIGatewayProxyEventHeaders,
+): StandardisedHeaders => {
+  const standardisedHeaders: StandardisedHeaders = {};
+  if (!apiGwHeaders) return standardisedHeaders;
+  const apiGwHeaderKeys = Object.keys(apiGwHeaders);
+  apiGwHeaderKeys.forEach((headerKey) => {
+    const headerValue = apiGwHeaders[headerKey];
+    if (
+      typeof headerValue === "string" ||
+      typeof headerValue === "number" ||
+      typeof headerValue === "boolean"
+    ) {
+      // This fails TLS validation as the target of the request (private apigw execute-url isn't on the certificate for this DNS)
```

Comment:
Failing TLS validation sounds bad. Are you saying the next line fixes it? Or is TLS validation going to fail and we're ok with that?

Reply:
bad wording on my part - will update! Effectively there are a bunch of AWS headers that are included automatically from the Proxy that are included - one causes a definite issue. We should probably try and strip all that we can. I'll update the comment and see whether any more headers can go